### DISCUSSION TITLE
option to disable kiai flashing in taiko

### DISCUSSION THREAD

**Discussion by Doksha** [ppy/osu#21040] [Ideas] [answered]
The new update causes hitcircles to flash during Kiai time, this is extremely distracting, and should have an option to disable. (if this also affects other gamemodes, the option should also apply, but I havent checked yet)

**Comment by eOliverr** [comment]
I agree. Not just because it's distracting, but the bright flashes are quite annoying in my opinion. Mapping already emphasizes music in much better ways. Kiai flashes just reduce enjoyment.

**Reply by WindowsMeosu** [reply]
Also seems like a feature that should be disabled by default since I doubt many people will actually prefer this to be enabled. I don't know how much of a risk it is for epilepsy, but I feel that's another reason for said setting to be disabled by default.

**Reply by Doksha** [reply]
unlikely, its a very minor brightness change, but it does make it a lot harder to read if you cant diferentiate colors and rely on brightness (which is already minisucle)
Its like hit lighting for standard, where the only player who even likes it is ekoro lol.

**Comment by vavalth** [comment]
Hello, samzies, I have stupid hypersensibility, I used the tweek of an old topic on the official osu forums by chaning the skin colors to match and never go bright even during kiai time. 

Last time I played was like a month ago, something like 28 days, right before that new update :( 

BUT, the new blinking is not a picture.jpg so I can't change it.

I can't play the game anymore, it's sad to say but it goes not further than that. 

In osu! it would all go away with the "Shaders" option, is that hard to add ? (I don't know heck about coding and stuff it's a legit question lol)

Just waving a flag of "Help me I can't play this awesome addictif game".

Lots of love !

**Comment by Joppe27** [comment]
I added these effects back as they are always on in stable and it makes the gameplay much less boring and more cheerful imo. I understand the concerns you raised though, but I'm unsure if we want to make this a setting since this wasn't configurable in stable either. Can a maintainer tell me if it's okay to make this a setting (maybe coupled with the edge effects that are in review because that will be an additional "distraction")? Will look into it if wanted.

**Reply by vavalth** [reply]
Hey there, thanks for your time. If you mean by "wasn't configurable in stable" that you could not "remove" the light in the "original" taiko, you could. I literally had everything off no lights no blur no blink. Kiai mode or not , that was the same for me.

**Reply by Joppe27** [reply]
It would be helpful if you could link me the forum post you were talking about (or explain yourself how you managed to hide these effects). Either way, I'm pretty sure this is not something that was officially supported.

**Reply by vavalth** [reply]
Hey there, site was the old osu forums , will be hard to keep the link again.

But in short ? Personalised skin. I think it's something like "defaultbar.jpg" (No clue at all for the names sorry lol) that .jpg is the bar where the dons and the kais appear and dissapear. the "kiaitimebar.jpg" is when it is the kiai time, thus, the image change : making it a glowing light buzzy image and in game the recreation is the kiai time full lights. What I did following that topic is literally copy the "defaultbar.jpg" and past it in my osu skin map while renaming it "kiaitimebar.jpg" so once the kiai time goes on, the game charges the "kiaitimebar.jpg" but it IS the same as the "no kiai time" so, no glow in the dark ! 

Edit : Here is a random link I found where they "explain" the same idea, it was common knowledge for those who are quickly affected by bright lights; https://www.reddit.com/r/osugame/comments/4amuta/kiai_burst_in_taiko_how_to_disable/

**Reply by Joppe27** [reply]
I assume you're talking about `taiko-bar-right` and `taiko-bar-right-glow`? These elements are already fully skinnable in lazer and can be configured to not do anything when kiai time starts using the same method you described for stable. The link you added also mentions the `taiko-glow` element which does not (yet) exist in lazer and does not do anything when kiai starts.

None of the things you described apply to kiai flashes, so we'll have to wait for an answer from the devs whether they want an option for that or not.

**Reply by vavalth** [reply]
Ok so I tought it was obvious that the only problem right now was the update you did. The thing YOU added is not linked to any skinable image, like the other glowing parts like you quoted "taiko bar right" and "taiko bar right glow" thus, I cannot remove what you did add. Everything was question of pictures before you added that new glow effect (stable also). My hole osu lazer skin is no-glow skin it did work perfectly before you did your update. If you did add that glow with a picture like the other glow parts, we would not have this conversation because I would have changed that picture in a no-glow one. So yeah I hope the devs are ok to remove or at least to let us change something that was changeable in the stable version.

**Reply by vavalth** [reply]
Yeah buddy; confusion emoji indeed. I see how I'm the crazy one making a big deal of it, but you have no idea how much that small light affects the hole thing for me. I've aspergers so I'm like a moth compared to you (or if you have aspergers you should know what I mean) when kiai starts my eyes go mad like little fairy's dancing left-right, it is impossible to focus. 

![Asperger_vs_normal_brain](https://user-images.githubusercontent.com/18542953/204036192-0f049787-71b2-48d5-b176-0df27d6f62c9.jpg)

Just so you have an idea ... would it be such a sin to add the option to disable it ? :(

**Comment by vavalth** [comment]
Any news ? Perhaps put it back how it was before untill a dev tells you to put the glow back again with an option to remove it like 90% of the users will do ?

**Reply by mk56-spn** [reply]
Devs ( maintainers ) approve every single thing that gets changed on this repo, if its in there its because they wanted it to be. id suggest patiently waiting for an update, prodding people for it will not make it happen any faster

**Reply by Joppe27** [reply]
@peppy could you tell me whether you want this to be configurable or not (and how as in option in menu or via skin) so this conversation can finally end? I'm fed up with it.

**Reply by vavalth** [reply]
> Devs ( maintainers ) approve every single thing that gets changed on this repo, if its in there its because they wanted it to be. id suggest patiently waiting for an update, prodding people for it will not make it happen any faster

Was just asking for any updates, sorry if it feld rude. Not really fan of the new aura neither.

**Comment by vernonlim** [comment]
I just tried out lazer properly due to hard links finally being added and while I'm not exactly sure why, I can say the flashing is way more distracting than it was in stable. Especially on maps that are higher bpm (250+), the flashes get so fast it legitimately makes it more difficult to play, and I don't have issues with light sensitivity. This definitely has to be toggleable in some form, or toned down.

I'm using a legacy skin, and from surface observation: I think the flashes on the Katsus seem brighter than they were in stable, and more contrasted to the Dons? Not sure if that's the case, but if the flashes were a lot dimmer, this issue would be a lot less of a problem.

EDIT: On Argon/Argon Pro the flashes seem to apply much more equally to the Kats/Dons, making it much less of a problem - although they still are bright enough to be really distracting at higher bpms.

EDIT 2: I'll attach videos showing the effect with Argon and Legacy skins in a moment.
Argon: https://streamable.com/b3h4qh
Legacy: https://streamable.com/vb4blc
Stable: https://streamable.com/wcf5r2

Looking at it this way, I actually think the effect is the most distracting with Argon, due to it being so visible.

Apologies for the background, only map I could think of with kiai at the beginning. Link is https://osu.ppy.sh/beatmapsets/1141971, and the bpm is 276.

**Comment by peppy** [comment, marked as answer]
#22081 #21569
